mantidproject/mantid
# Fix progress reporting in LoadNexusProcessed

Description of work.

This fixes progress reporting the LoadNexusProcessed so that updates incrementally within a parallel loop. I also updated a few variables in `ProgressBase` to be atomic and added a check to ensure that the end variables is <= 1.0. The scope of locks in `EventWorkspaceMRU::deleteIndex` have been reduced. Most of the other changes just avoid implicit casts between integer and double types. 

Inside MantidPlot, this significantly speeds up loading files. Loading `/SNS/CORELLI/IPTS-17467/shared/autoreduce/CORELLI_34682_elastic.nxs` decreased from 12.86 to 3.17 seconds. From python (no progress reporting), the benefits are considerably less. 

**To test:**



 * Code review. 
 * Verify LoadNexus now correctly reports progress from 0 to 1
 * Verify performance improvement running LoadNexus inside MantidPlot. 

There is no GitHub issue associated with this pull request

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
